input/script_utils.go

```diff
@@ -2060,6 +2096,14 @@ type CommitScriptTree struct {
 	// RevocationLeaf is the leaf used to spend the output with the
 	// revocation key signature.
 	RevocationLeaf txscript.TapLeaf
+
+	// AuxLeaf is an axuillaruy leaf that can be used to extend the base
+	// commitment script tree with new spend paths, or just as extra
+	// commitment space. When present, this leaf will always be in the
+	// left-most or right-most area of the tapscript tree.
+	//
+	// TODO(roasbeeF): need to ensure commitment position...
```

Comment:
Is this TODO related to my comment above? About relying on the logic in `txscript.AssembleTaprootScriptTree`?

Reply:
Yep, but as it it works since here we have it be that odd element out, which'll give it the top right position in the tree.